npm@5

npm@5 was released with lots of nice features and better performance.

http://blog.npmjs.org/post/161276872334/npm5-is-now-npmlatest

Does it make sense to update it in `stable`?

@harti2006 No, we won't because the issue happens when we install `npm@5` after installing Node.js. It won't happen **somehow** when we install `npm@5` in the same `RUN` as Node.js installation.

I tested it on my local and it works, but I had to change `npm cache clear` to `npm cache clean --force` because of the following error. (`npm cache clear` is an alias for `npm cache clean`.)

```
npm ERR! As of npm@5, the npm cache self-heals from corruption issues and data extracted from the cache is guaranteed to be valid. If you want to make sure everything is consistent, use 'npm cache verify' instead.
npm ERR!
npm ERR! If you're sure you want to delete the entire cache, rerun this command with --force.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2017-06-14T15_43_06_856Z-debug.log
```

The reason for it is documented in https://docs.npmjs.com/cli/cache#details

--- a/stable/Dockerfile
+++ b/stable/Dockerfile
@@ -25,7 +25,7 @@ RUN set -ex \
 
 ENV NPM_CONFIG_LOGLEVEL info
 ENV NODE_VERSION 7.10.0
-ENV NPM_VERSION 5.0.1
+ENV NPM_VERSION 5.0.3
 
 RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
   && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \

--- a/stable/Dockerfile
+++ b/stable/Dockerfile
@@ -25,7 +25,7 @@ RUN set -ex \
 
 ENV NPM_CONFIG_LOGLEVEL info
 ENV NODE_VERSION 7.10.0
-ENV NPM_VERSION 4.5.0
+ENV NPM_VERSION 5.0.1
 
 RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
   && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \
@@ -34,6 +34,6 @@ RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-
   && tar -xzf "node-v$NODE_VERSION-linux-x64.tar.gz" -C /usr/local --strip-components=1 \
   && rm "node-v$NODE_VERSION-linux-x64.tar.gz" SHASUMS256.txt.asc SHASUMS256.txt \
   && npm install -g npm@"$NPM_VERSION" \
-  && npm cache clear
+  && npm cache clean --force
 
 CMD [ "node" ]

--- a/stable/Dockerfile
+++ b/stable/Dockerfile
@@ -25,7 +25,7 @@ RUN set -ex \
 
 ENV NPM_CONFIG_LOGLEVEL info
 ENV NODE_VERSION 7.10.0
-ENV NPM_VERSION 5.0.3
+ENV NPM_VERSION 5.1.0
 
 RUN curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-x64.tar.gz" \
   && curl -SLO "https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc" \

